src/schema/workflow_execution_activity.yaml

```diff
@@ -44,7 +45,47 @@ classes:
       - part_of
       - type
       - version
-
+      - qc_status
+      - qc_comment
+      - has_failure_categorization
+    rules:
+      - title: qc_status_pass_has_output_required
+        description: >-
+          If qc_status has a value of pass, then the has_output slot is required.
+        preconditions:
+          slot_conditions:
+            qc_status:
+              equals_string: pass
+        postconditions:
+          slot_conditions:
+            has_output:
+              required: true
+      - title: qc_status_pass_null_has_output_required
+        description: >-
+          If qc_status is not specified, then the has_output slot is required.
+        preconditions:
+          slot_conditions:
+            qc_status:
+              value_presence: ABSENT
+        postconditions:
+          slot_conditions:
+            has_output:
+              required: true
+
+      # - title: qc_status_pass_qc_where_what_slots_not_specified
```

Comment:
are we going to leave this in as commented-out code?

Reply:
Deleted